packages/smart-wallet/src/typeGuards.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+
+import { M } from '@agoric/store';
+import {
+  InstanceHandleShape,
+  ProposalShape,
+} from '@agoric/zoe/src/typeGuards.js';
+
+export const shape = {
+  // smartWallet
+  StringCapData: {
+    body: M.string(),
+    slots: M.arrayOf(M.string()),
+  },
+
+  // invitations
+  ContractInvitationSpec: M.split(
+    {
+      source: 'contract',
+      instance: InstanceHandleShape,
+      publicInvitationMaker: M.string(),
+    },
+    M.partial({
+      invitationArgs: M.array(),
+    }),
+  ),
+  ContinuingInvitationSpec: M.split(
+    {
+      source: 'continuing',
+      previousOffer: M.number(),
+      invitationMakerName: M.string(),
+    },
+    M.partial({
+      invitationArgs: M.array(),
+    }),
+  ),
+  PurseInvitationSpec: {
+    source: 'purse',
+    instance: InstanceHandleShape,
+    description: M.string(),
+  },
+
+  // offers
+  OfferShape: M.split(
+    {
+      id: M.number(),
+      // TODO M.unknown() to defer validation
+      invitationSpec: M.any(),
+      // XXX ProposalShape requires all whereas TS types ProposalRecord does and Proposal is partial
```

Comment:
Can delete the XXX? This perfectly reasonable.

Reply:
Unless you want to recommend renaming the current `ProposalShape` to `FullProposalShape` and publishing the below as `ProposalShape`?

Reply:
right I propose a rename. that particular rename SGTM. sounds like I have your support so I'll do that in this PR>